File: MiniDOMBase/DMD/Anagrafica/Banche/BBAN/CBBAN.cs
using System;
using DMD;
using DMD.Databases;
using static minidom.Sistema;

namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Classe che rappresenta un codice BBAN
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CBBAN 
            : DMD.XML.DMDBaseXMLObject
        {

            private string m_Codice;

            /// <summary>
            /// Costruttore
            /// </summary>
            public CBBAN()
            {
            }

            /// <summary>
            /// Costruttore
            /// </summary>
            /// <param name="codice"></param>
            public CBBAN(string codice) : this()
            {
                m_Codice = DMD.Strings.Trim(codice);
            }

            /// <summary>
            /// Restituisce o imposta il codice
            /// </summary>
            public string Codice
            {
                get
                {
                    return m_Codice;
                }

                set
                {
                    value = DMD.Strings.UCase(DMD.Strings.Left(DMD.Strings.Trim(value), 2));
                    string oldValue = m_Codice;
                    m_Codice = value;
                    //DoChanged("Codice", value, oldValue);
                }
            }

            /// <summary>
            /// Restituisce una stringa che rappresenta l'oggetto
            /// </summary>
            /// <returns></returns>
            public override string ToString()
            {
                return this.m_Codice;
            }

            /// <summary>
            /// Restituisce vero se il codice rappresentato è valido
            /// </summary>
            /// <returns></returns>
            /// <remarks></remarks>
            public virtual bool IsValid()
            {
                return true;
            }

            //protected internal override DBConnection GetConnection()
            //{
            //    return Databases.APPConn;
            //}

            

            ///// <summary>
            ///// Restituisce il discriminante
            ///// </summary>
            ///// <returns></returns>
            //public override string GetTableName()
            //{
            //    return "tbl_BBANs";
            //}

            /// <summary>
            /// Restituisce il codice hash dell'oggetto
            /// </summary>
            /// <returns></returns>
            public override int GetHashCode()
            {
                return HashCalculator.Calculate(this.m_Codice);
            }

            /// <summary>
            /// Restituisce true se i due oggetti sono uguali
            /// </summary>
            /// <param name="obj"></param>
            /// <returns></returns>
            public sealed override bool Equals(object obj)
            {
                return (obj is CBBAN) && this.Equals( (CBBAN)obj);
            }

            /// <summary>
            /// Restituisce true se i due oggetti sono uguali
            /// </summary>
            /// <param name="obj"></param>
            /// <returns></returns>
            public virtual bool Equals(CBBAN obj)
            {
                return //base.Equals(obj) && 
                    DMD.Strings.EQ(this.m_Codice, obj.m_Codice);
            }
        }
    }
}

File: MiniDOMBase/DMD/Anagrafica/Banche/BBAN/CBBAN.vb
Partial Public Class Anagrafica

    ''' <summary>
    ''' Classe che rappresenta un codice BBAN
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CBBAN
        Inherits minidom.Databases.DBObjectBase

        Private m_Codice As String

        Public Sub New()
        End Sub

        Public Sub New(ByVal codice As String)
            Me.New()
            Me.m_Codice = Trim(codice)
        End Sub

        Public Property Codice As String
            Get
                Return Me.m_Codice
            End Get
            Set(value As String)
                value = UCase(Left(Trim(value), 2))
                Dim oldValue As String = Me.m_Codice
                Me.m_Codice = value
                Me.DoChanged("Codice", value, oldValue)
            End Set
        End Property

        Public Overrides Function ToString() As String
            Return Me.m_Codice
        End Function

        ''' <summary>
        ''' Restituisce vero se il codice rappresentato è valido
        ''' </summary>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overridable Function IsValid() As Boolean
            Return True
        End Function

        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return minidom.Databases.APPConn
        End Function

        Public Overrides Function GetModule() As Sistema.CModule
            Return Nothing
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_BBANs"
        End Function
    End Class

End Class